src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -149,13 +150,22 @@ public void Dispose()
 
         private void Reset(bool isDisposing = false)
         {
-            _workerCts?.Cancel();
+            if (_workerCts?.IsCancellationRequested == false)
+            {
+                _workerCts.Cancel();
```

Comment:
You can call `.Cancel()` multiple times, the CTS should return if it is already been requested. However, it will throw if the CTS is disposed. So you probably want to change it to `_workerCts?.IsDisposed == false`.